internal/hub/config.go

```diff
@@ -67,7 +67,7 @@ func (c *Config) Sanitize() {
 	c.Key = strings.ToLower(c.Key)
 
 	if c.Secret != "" {
-		if c.Key != fmt.Sprintf("%x", sha1.Sum([]byte(c.Secret))) {
+		if c.Key != fmt.Sprintf("%x", sha512.New512_256().Sum([]byte(c.Secret))) {
```

Comment:
I think long term this area should move away from using hashing completely. Ideally you'd want to use something like PBKDF2 or, preferably, bcrypt/scrypt. A single round of hashing without a salt is too weak.

Reply:
If secrets are being checked here you can't just swap out the hash mechanism with a different one without providing a migration path for previously stored secrets

Reply:
Hoping one of the author's can confirm my understanding of this area. If it is the case then I'd need some direction on providing a migration path as I'm not familiar with the majority of the codebase.

Reply:
We don't need higher security at the moment as the main use is distributing maps keys. We pay for them, so we wan't to avoid simple abuse (like script kids getting the keys in plain text using a perl script).